src/main/java/dev/blaauwendraad/masker/json/ValueMaskers.java

```diff
@@ -169,25 +169,67 @@ public static ValueMasker.StringMasker email(int keepPrefixLength, int keepSuffi
     }
 
     /**
-     * Masks a target value with the provided {@link Function}. The target value is passed into the
-     * function as a text, regardless of the type (string, numeric or a boolean), however any
-     * non-null returned value from the function will always be a JSON string (with quotes added
-     * automatically).
+     * Masks a target value with the provided {@link Function}. The target value (as raw JSON literal) is passed into
+     * the function as a string regardless of the JSON type (string, numeric or a boolean). In case the target value is
+     * a JSON string the value the function will receive a JSON encoded value with the quotes as it appears in the JSON
+     * with line breaks converted to \n, special characters like " or \ escaped with a backslash (\).
```

Comment:
```suggestion
     * with line breaks encoded as  \n, special characters like " or \ escaped with a backslash (\).
```

Reply:
Those as well and there are quite some more + all unicode, so I didn't want to include them all

Reply:
Then I would mention that that is just an example

Reply:
Darn, it got automerged because committing a suggestion resolves the thread 🙃

Reply:
😂😂